assets/src/scripts/monitoring-location/components/hydrograph/parameters.js

```diff
@@ -17,109 +15,218 @@ import {getAvailableParameters} from './selectors/parameter-data';
-                    inputElem.property('checked', d => d.parameterCode === selectedParameterCode ? true : null);
-                }, getSelectedParameterCode));
-            paramSelectCol.append('label')
-               .attr('class', 'usa-radio__label');
-            const paramCdCol = tr.append('th')
-                .attr('scope', 'row');
-            paramCdCol.append('span')
-                .text(d => d.description)
-                .each(function(d) {
-                    appendInfoTooltip(select(this), `Parameter code: ${d.parameterCode}`);
-                });
-            tr.append('td')
-                .style('white-space', 'nowrap')
-                .text(d => d.periodOfRecord ?
-                    `${d.periodOfRecord.begin_date} to ${d.periodOfRecord.end_date}` : '');
-            tr.append('td')
-                .append('div')
-                .attr('class', 'wateralert-link');
         });
+};
 
-    // WaterAlert does not support every parameter code, so lets take that into account when adding the links
-    table.selectAll('.wateralert-link').each(function(d) {
-        let selection = select(this);
+/*
+* Helper function that creates the top row of each parameter selection. This row is hidden except on narrow screens
+* and contains the period of record that appears above the parameter description.
+* @param {Object} Element - The target element to append the row
+* @param {Object} parameter - Contains details about the current parameter code
+* @return {Object} The HTML for the grid row
+* */
+const drawTopPeriodOfRecordRow = function(element, parameter) {
+    const gridRowInnerTopPeriodOfRecord = element.append('div')
+        .attr('class', 'grid-row grid-row-inner grid-row-period-of-record');
+    gridRowInnerTopPeriodOfRecord.append('div')
+        .attr('class', 'grid-col-10 grid-offset-1')
+        .text(`${parameter.periodOfRecord.begin_date} to ${parameter.periodOfRecord.end_date}`);
+    const TopPeriodOfRecordRowExpansionControlDiv = gridRowInnerTopPeriodOfRecord.append('div')
+        .attr('class', 'grid-col-1 open-close-top-period-of-record');
 
-        let waterAlertCell;
-        if (d.waterAlert.hasWaterAlert) {
-            waterAlertCell = selection.append('a')
-                .attr('href', `${config.WATERALERT_SUBSCRIPTION}/?site_no=${siteno}&parm=${d.waterAlert.subscriptionParameterCode}`);
-        } else {
-            waterAlertCell = selection;
-        }
-        waterAlertCell.attr('class', 'water-alert-cell usa-tooltip')
+    drawRowExpansionControl(TopPeriodOfRecordRowExpansionControlDiv, parameter, 'mobile');
+};
+
+/*
+* Helper function that draws the row containing the radio button and parameter description.
+* @param {Object} Store - The application Redux state
+* @param {Object} Element - The target element to append the row
+* @param {Object} parameter - Contains details about the current parameter code
+*/
+const drawRadioButtonRow = function(store, element, parameter) {
```

Comment:
Our pattern has been to list the "element" parameter first, giving the caller the option of using the D3 .call function.

Reply:
Interesting, I thought about that before. Thanks for pointing out; will change.